src/components/calcite-list-item/calcite-list-item.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, Element, Event, EventEmitter, Prop, h, VNode } from "@stencil/core";
+import { SLOTS, CSS } from "./resources";
+import { getSlotted } from "../../utils/dom";
+
+/**
+ * @slot actions-start - todo document.
+ * @slot content-start - todo document.
+ * @slot content-end - todo document.
+ * @slot actions-end - todo document.
+ */
+@Component({
+  tag: "calcite-list-item",
+  styleUrl: "calcite-list-item.scss",
+  shadow: true
+})
+export class CalciteListItem {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * @todo document.
+   */
+  @Prop({ reflect: true }) buttonDisabled = false;
```

Comment:
Is there another name to convey what this mode does? I think `button` is an implementation detail.

Reply:
We had discussed this in the issue. https://github.com/Esri/calcite-components/issues/2117

Do you have a better name in mind?

Reply:
I'd have to mull for a bit, but off the top of my head `interactive`?

Reply:
So `non-interactive` maybe? Because the default would be interactive unless we want to opt into it.